editor/store/reducer.js

```diff
@@ -584,6 +584,26 @@ export function isTyping( state = false, action ) {
 	return state;
 }
 
+/**
+ * Reducer returning autosave state. True if an autosave is currently occurring.
+ *
+ * @param  {boolean} state  Current state.
+ * @param  {Object}  action Dispatched action.
+ *
+ * @return {boolean} Updated state.
+ */
+export function isAutosaving( state = false, action ) {
```

Comment:
```js
it( 'should return ??? when the post is published while autosave is in progress', () => {
	const original = isAutosaving( undefined, savePost( { autosave: true } ) );
	const state = isAutosaving( original, savePost() );

	expect( state ).toBe( /* ??? */ );
} );
```

Note: UI limitations imposed aren't enough. If we want it to be a case prevented, at the very least it should be considered as part of an `isEditedPostPublishable` selector or equivalent state layer consideration (e.g. considered as part of the `REQUEST_POST_UPDATE` effect).

It maybe speaks to a broader concern of: Is having separate state handling for "is autosaving" vs. "is saving" causing us more headaches to handle than if they were one in the same? Does it make sense for them to be considered as equivalent?

Reply:
possibly. autosaves and saves are very distinct and an autosave could be happening and then a save fired, these shouldn't interfere with each other.  ideally, you should be able to click the update button even while an autosave is firing?

if you are concerned about race conditions between the two, we could drop isAutosaving and prevent saves during autosaves and autosaves during saves.